go-controller/pkg/networkmanager/network_controller.go

```diff
@@ -240,19 +247,44 @@ func (c *networkController) syncAll() error {
 	// aware of all the existing networks on initialization. To achieve that, we need to start existing
 	// networks synchronously. Otherwise, these controllers might incorrectly assess valid configuration
 	// as stale.
-	start := time.Now()
-	klog.Infof("%s: syncing all networks", c.name)
+	eg, cancel := errgroup.WithContext(context.Background())
+	// start 1 network at a time
+	eg.SetLimit(1)
 	for _, network := range validNetworks {
-		err := c.syncNetwork(network.GetNetworkName())
-		if errors.Is(err, ErrNetworkControllerTopologyNotManaged) {
-			klog.V(5).Infof("Ignoring network %q since %q does not manage it", network.GetNetworkName(), c.name)
-			continue
-		}
-		if err != nil {
-			return fmt.Errorf("failed to sync network %s: %w", network.GetNetworkName(), err)
-		}
+		networkName := network.GetNetworkName()
+		eg.Go(func() error {
+			err := retry.OnError(
+				wait.Backoff{
+					// retry each network every 500ms during 1m
+					Steps:    120,
+					Duration: 500 * time.Millisecond,
+					Factor:   1.0,
+					Cap:      1 * time.Minute,
+				},
+				func(err error) bool {
+					select {
+					case <-cancel.Done():
+						// at least one network failed to start, no point in
+						// retrying further
+						return false
+					default:
+						return !errors.Is(err, ErrNetworkControllerTopologyNotManaged)
+					}
+				},
+				func() error { return c.syncNetwork(networkName) },
+			)
+			if errors.Is(err, ErrNetworkControllerTopologyNotManaged) {
+				klog.V(5).Infof("Ignoring network %q since %q does not manage it", networkName, c.name)
+				return nil
+			}
+			return err
+		})
```

Comment:
Neat! I like your solution. This also addresses this bug of mine: https://issues.redhat.com/browse/OCPBUGS-45239

Reply:
@ricky-rav can you explain how this addresses your issue? This will still crash-loop if the attempts to start a network fail. This is just moving the retry logic from the factory to here.